ts/main.ts

```diff
@@ -1,66 +1,326 @@
+      },
+    ],
+  },
+};
+
+const currentGridId: number = 1;
+const gridSize: number = 3;
+let numCorrect: number = 0;
+
+const $topCategories = document.querySelectorAll(
+  '.top-cat',
+) as NodeListOf<HTMLDivElement>;
+if (!$topCategories) throw new Error('.top-cat query failed!');
+const $leftCategories = document.querySelectorAll(
+  '.left-cat',
+) as NodeListOf<HTMLDivElement>;
+if (!$leftCategories) throw new Error('left-cat query failed!');
+
+function renderGridCategories(gridID: number): void {
+  for (let i = 0; i < gridSize; i++) {
+    const imgContainerTop: HTMLDivElement = document.createElement('div');
+    imgContainerTop.classList.add('w-1/2', 'h-1/2', 'flex', 'justify-center');
+    const catImgTop: HTMLImageElement = document.createElement('img');
+    catImgTop.setAttribute(
+      'src',
+      categorySets[gridID].topCategories[i].imgPath,
+    );
+    imgContainerTop.appendChild(catImgTop);
+    $topCategories[i].appendChild(imgContainerTop);
+    const catText: HTMLHeadingElement = document.createElement('h1');
+    catText.textContent = categorySets[gridID].topCategories[i].text;
+    $topCategories[i].appendChild(catText);
+
+    const imgContainerLeft: HTMLDivElement = document.createElement('div');
+    imgContainerLeft.classList.add('w-1/2', 'h-1/2', 'flex', 'justify-center');
+    const catImgLeft: HTMLImageElement = document.createElement('img');
+    catImgLeft.setAttribute(
+      'src',
+      categorySets[gridID].leftCategories[i].imgPath,
+    );
+    imgContainerLeft.appendChild(catImgLeft);
+    $leftCategories[i].appendChild(imgContainerLeft);
+    const catTextLeft = document.createElement('h1');
+    catTextLeft.textContent = categorySets[gridID].leftCategories[i].text;
+    $leftCategories[i].appendChild(catTextLeft);
+  }
+} // function that renders Grid Categories based on what number Grid
+
+renderGridCategories(currentGridId);
+
+const $gridContainer = document.querySelector(
+  '.grid-container',
+) as HTMLDivElement;
+if (!$gridContainer) throw new Error('.grid-container query failed!');
+const $modal = document.querySelector('.modal') as HTMLDivElement;
+if (!$modal) throw new Error('.modal query failed!');
+const $input = document.querySelector('input') as HTMLInputElement;
+if (!$input) throw new Error('input query failed');
+
+$gridContainer.addEventListener('click', function (event: Event) {
+  const element = event.target as HTMLElement;
+  console.log(element.classList);
+  if (
+    element.classList.contains('game-square') &&
+    !element.classList.contains('correct')
+  ) {
+    console.log('game sq!');
+    $modal.classList.remove('hidden');
+    element.classList.add('bg-yellow-500', 'selected-sq');
+  }
+});
+
+$modal.addEventListener('click', function (event: Event) {
+  const element = event.target as HTMLElement;
+  const $selectedSq = document.querySelector('.selected-sq') as HTMLDivElement;
+  if (!$selectedSq) throw new Error('.selected-sq query failed!');
```

Comment:
What are you doing with the element after you query for it?

Reply:
im using it as an argument for my clearSelectedSqResetModal() function call so that it is cleared of being a selected square and goes back to having a white background color